client/varsdir.c

```diff
@@ -49,12 +50,11 @@ struct cnfnode *parse_varsdirs(char *dirs[])
                 continue;
             }
 
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-truncation"
-            snprintf(path, sizeof(path), "%s/%s", dirs[i], dent->d_name);
-#pragma GCC diagnostic pop
-#endif /* __GNUC__ && !__clang__ */
+            n = snprintf(path, sizeof(path), "%s/%s", dirs[i], dent->d_name);
+            if (n < 0 || n >= (int)sizeof(path)) {
+                errno = ENAMETOOLONG;
```

Comment:
If `n < 0` then the error is something else. `snprintf()` should have set `errno` in that case, so do not overwrite it.